flag-app/flag-app-project/src/components/CardWrapper.tsx

```diff
@@ -1,21 +1,27 @@
 import { useCountries } from "../hooks";
+import { Country } from "../interfaces";
 import Card from "./Card";
+import { CardSkeleton } from "./CardSkeleton";
 import ErrorMessage from "./ErrorMessage";
 
 interface Props {
-  searchInput: string;
+  handleCardClick: (country: Country) => void;
 }
 
-export const CardWrapper = ({ searchInput }: Props) => {
-  const { countries, error, isLoading } = useCountries(searchInput);
+export const CardWrapper = ({ handleCardClick }: Props) => {
+  const { countries, error, isLoading } = useCountries();
 
-  if (isLoading) return <p>Loading...</p>;
-  if (error) return <ErrorMessage searchInput={searchInput} />;
+  if (error) return <ErrorMessage />;
 
   return (
     <div className="cards-layout">
-      {countries.map((country) => (
-        <Card country={country} key={country.cca3} />
+      {countries?.map((country) => (
```

Comment:
Do an early return if there are no `countries` above L15, just like for `isLoading` or `error`.